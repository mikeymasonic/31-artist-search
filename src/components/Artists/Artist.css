@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@200;300&display=swap');

.artist {
  height: 10vw;
  width: 20vw;
  text-align: center;
  margin: 0 3vw 3vw 0;
  border: .2vw solid rgb(133, 220, 226);
  border-radius: 1rem;
  font-size: 1.4vw;
  font-family: 'Montserrat', sans-serif;
}

.artist span {
  color: gray;
  font-size: 1vw;
}

.link {
  text-decoration: none;
  color: #f7d89e;
  font-weight: 200;
  font-family: 'Montserrat', sans-serif;
}

.link:hover {
  color: white;
}

.artist:hover {
  background: hotpink;;
}

.titleh2 {
  font-size: 2vw;
  font-family: 'Montserrat', sans-serif;
  text-align: center;
  color: #f7d89e;
  font-weight: 200;
}