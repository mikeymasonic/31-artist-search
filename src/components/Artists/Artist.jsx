import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Artist = ({ artistId, artistName }) => {
  return (
    <>
      <Link to={`/releases/${artistName}/${artistId}`}><h3>{artistName}</h3>
      </Link>
    </>  
  );

};

Artist.propTypes = {
  artistId: PropTypes.string.isRequired,
  artistName: PropTypes.string.isRequired
};

export default Artist;
