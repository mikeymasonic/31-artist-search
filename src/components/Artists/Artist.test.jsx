import React from 'react';
import { shallow } from 'enzyme';
import Artist from './Artist';

describe('Artists component', () => {
  it('renders Artist', () => {
    const wrapper = shallow(<Artist artistId='311' artistName='Silkworm' />);
    expect(wrapper).toMatchSnapshot();
  });
});
