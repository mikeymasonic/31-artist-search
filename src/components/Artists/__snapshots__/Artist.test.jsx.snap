// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists component renders Artist 1`] = `
<Fragment>
  <Link
    className="link"
    to="/releases/Silkworm/311"
  >
    <section
      className="artist"
    >
      <h3
        className="titleh2"
      >
        Silkworm
      </h3>
      <span />
    </section>
  </Link>
</Fragment>
`;
